core-tests/shared/src/test/scala/zio/sharedLayers/SharedLayer.scala

```diff
@@ -0,0 +1,19 @@
+package zio.sharedLayers
+
+import zio.{Has, ZIO, ZLayer}
+
+import java.util.concurrent.atomic.AtomicInteger
+
+object SharedLayer {
+  private val counter = new AtomicInteger(0)
+
+  val test: ZLayer[Any, Nothing, Has[Unit]] = {
+    ZLayer.fromZIO(ZIO.debug("Constructing a super expensive layer") *> ZIO.succeed(updateCounter()))
```

Comment:
We should probably remove this debug statement.

Reply:
Yep, removed